compiler-rt/lib/lsan/lsan_interceptors.cpp

```diff
@@ -77,6 +77,8 @@ INTERCEPTOR(void*, malloc, uptr size) {
 }
 
 INTERCEPTOR(void, free, void *p) {
+  if (!p)
```

Comment:
this will fix only lsan

to fix all, I propose to change 
`DlsymAlloc::PointerIsMine` to return true for NULL
and teach `DlsymAlloc::Free` to handle it correctly

Reply:
I see, you are correct.

Can you please include all other sanitizers here, also cfree, and please use UNLIKELY(!p)

Reply:
Please rebase and include into the patch removal of XFAIL added in #108289